libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -238,64 +244,158 @@ string IRGenerator::generateInitialAssignment(VariableDeclaration const& _varDec
 	return generator.code();
 }
 
-string IRGenerator::constructorCode(ContractDefinition const& _contract)
+pair<string, map<ContractDefinition const*, string>> IRGenerator::evaluateConstructorArguments(
+	ContractDefinition const& _contract
+)
 {
-	// Initialization of state variables in base-to-derived order.
-	solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
-
-	using boost::adaptors::reverse;
-
-	ostringstream out;
+	IRGeneratorForStatements generator{m_context, m_utils};
+	auto listParamNames = [&](
+		FunctionDefinition const* baseConstructor,
+		vector<ASTPointer<Expression>> const* arguments
+	) -> string
+	{
+		string params;
+		for (size_t i = 0; i < arguments->size(); ++i)
+			params += (params.empty() ? "" : ", ") + generator.evaluateExpression(
+				*(arguments->at(i)),
+				*(baseConstructor->parameters()[i]->type())
+			).commaSeparatedList();
+
+		return params;
+	};
+
+	map<ContractDefinition const*, string> constructorParams;
+	if (FunctionDefinition const* constructor = _contract.constructor())
+		for (auto const& modifier: constructor->modifiers())
```

Comment:
Can you combine this loop with the next using a vector of pairs of contract definitions and arguments?

Reply:
I did combine to create vector of pairs than execute same on every pair in vector, hope that was the thing you suggested here :)